### PR TITLE
retire the connection ID sent in the preferred_address transport parameter

### PR DESCRIPTION
**Pull request by marten-seemann**

### CODE REVIEW

File: example/client/main.go [hunk 1]
```diff
@@ -1,75 +1,68 @@
...
-	"crypto/x509"
-	"flag"
-	"io"
-	"net/http"
-	"sync"
-
-	"github.com/lucas-clemente/quic-go/http3"
-	"github.com/lucas-clemente/quic-go/internal/testdata"
-	"github.com/lucas-clemente/quic-go/internal/utils"
+	"fmt"
```
Comment (lucas-clemente) [inline example/client/main.go]: Probably revert the changes to this file?

Comment (marten-seemann) [reply example/client/main.go]: 🤦🏻‍♂
